pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -75,13 +76,42 @@ func TestReconcile(t *testing.T) {
 		expectRequeueAfter     time.Duration
 	}{
 		{
-			name: "cluster deleted",
-			cd:   cdBuilder.GenericOptions(testgeneric.Deleted()).Options(o.shouldHibernate).Build(),
+			name: "cluster deleted with finalizer",
+			cd:   cdBuilder.GenericOptions(testgeneric.Deleted(), testgeneric.WithFinalizer(finalizer)).Options(o.shouldHibernate).Build(),
 			cs:   csBuilder.Build(),
 			validate: func(t *testing.T, cd *hivev1.ClusterDeployment) {
 				cond, runCond := getHibernatingAndRunningConditions(cd)
-				assert.Nil(t, cond, "not expecting hibernating condition")
-				assert.Nil(t, runCond, "not expecting running condition")
+				require.NotNil(t, cond)
+				require.NotNil(t, runCond)
+				assert.Equal(t, corev1.ConditionUnknown, cond.Status)
```

Comment:
You can also do
```
if assert.NotNil(t, cond, "<A helpful debug msg>"){
    assert.Equal(t, <status>, cond.Status, "<A helpful debug msg>"}
```
There's also [helper functions](https://github.com/openshift/hive/blob/master/pkg/test/assert/assertions.go#L68:L108) if they make the checks simpler